modules/ROOT/pages/distributed-locking.adoc

```diff
@@ -0,0 +1,104 @@
+Refer to the following sample code which declares an extension operation that manages code access through Mule locks and thus prevents synchronization issues when the operation is executing in parallel (like in a scatter gather scope or in different cluster nodes). +
+. Use an identifier for the lock you create. +
+The identifier allows you to access the same lock from different threads without having to explicitly share the same lock instance between them. +
+
+This lock implements the `java.util.concurrent.Lock` interface, so calling the method `lock` will make the thread wait until it acquires the lock. +
+
+After obtaining the lock, the critical code section can be executed and, no matter what the result is, we want to release the obtained lock (to prevent deadlocks on other threads waiting for this lock), so we wrap the `lock.unlock` call in a `finally` block.
+
+[source,java,linenums]
+----
+public class ClusterOperations {
+
+  @Inject
+  private LockFactory lockFactory;
+
+  public void sharedResourceOperation() {
+      Lock lock = lockFactory.createLock("sharedResourceId");
+      lock.lock();
+      try {
+          // Lock acquired, execute critical code.
+      } finally {
+          lock.unlock();
+      }
+  }
+}
+----
+
+If you are using the scripting extension, you must first access the registry to obtain the `LockFactory` and then create locks through scripting code. The following example uses Groovy to achieve the same result as the previous example: obtain the  `LockFactory` (this time through the `Registry`), create a new `Lock` for the given ID, wait until obtaining the lock to execute critical code and after that release the lock.
+
+[source,groovy,linenums]
+----
+lockFactory = registry.lookupByName("_muleLockFactory").get()
+lock = lockFactory.createLock("sharedResourceId")
+lock.lock()
+try {
+	// Lock acquired, execute critical code.
+} finally {
+	lock.unlock()
+}
+----
+
+== Sharing a Lock Between Operations
+
+Locking is not only useful to create locks for many threads running the same operation, but it can also be used for different operations that may be running in parallel. Using the same lock ID to create a lock in different operations, will make them obtain access to the same lock. The following code sample describes two custom operations which create locks using the same lock ID.
+
+[source,java,linenums]
+----
+public class ClusterOperations {
+
+  @Inject
+  private LockFactory lockFactory;
+
+  public void sharedResourceOperationA() {
+      Lock lock = lockFactory.createLock("sharedResourceId");
+      lock.lock();
+      try {
+          // Lock acquired, execute critical code.
+      } finally {
+          lock.unlock();
+      }
+  }
+
+  public void sharedResourceOperationB() {
+      Lock lock = lockFactory.createLock("sharedResourceId");
+      lock.lock();
+      try {
+          // Lock acquired, execute critical code.
+      } finally {
+          lock.unlock();
+      }
+  }
+}
+----
+
```

Comment:
I wonder if this warning text should be specified at the beginning of the topic set, so that the audience is aware of this issue from the beginning and does not expect to use the `newCondition()` method? Or at the beginning of only the sharing topic?
I think Requirements and Restrictions are sort of like Prerequisites, and that audiences should know about them from the beginning, so they can choose whether or not to read on, if their environment does not support what is going to be documented.